src/app/[lang]/(user)/layout.tsx

```diff
@@ -0,0 +1,22 @@
+import "../globals.css";
+import { Navigation } from "@/components/navigation/index";
+import { Footer } from "@/components/footer/index";
+import { Locale } from "../../../../i18n.config";
+import { getDictionary } from "@/lib/dictionary";
+
+export default async function UserLayout({
+  children,
+  params,
+}: {
+  children: React.ReactNode;
+  params: { lang: Locale };
+}) {
+  const translations = await getDictionary(params.lang);
+  return (
+    <>
+      <Navigation translations={translations.navigation} lang={params.lang} />
```

Comment:
There's no need to pass all translations to Navigation. You always can access lang from any component using `useParams`:

```tsx
const { lang } = useParams()
const translations = await getDictionary(lang);
```

You can do this in any server component

Reply:
But that's ok also, your implementation will be more optimized, as you make request only once here